jep/0000/README.adoc

```diff
@@ -0,0 +1,232 @@
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+
+== Abstract
+
+Integral to link:https://github.com/jenkinsci/jep/tree/master/jep/300[Jenkins
+Essentials] is the client's update lifecycle with the Evergreen hosted service
+layer. Starting for the first boot, the client must frequently inform the
+client of its current versions, retrieve new versions, and apply updates. This
+document specifies those API interactions and expectations to allow this update
+lifecycle to be performed correctly.
+
+== Specification
+
+The role of the versions service is simply to maintain an audit trail of sorts
+for the versions of sofware that each client has installed locally.
+
+
+The role of the update service is to use the `essentials.yaml` and retriev the
+latest versions of software for a given client UUID and compute the appropriate
+update response by the client
+
+.Bootstrapping Instance
+[source]
+----
+
+  update                     versions                    evergreen
+  service                    service                      client
+     |                           |                           |
+     |                           |     POST version manifest |
+     |                           |     from local hpis/core  |
+     |                           |<--------------------------o
+     |                           o-----------/ok/----------->|
+     |                           |                           |
+     |                     GET /update                       |
+     |<------------------------------------------------------o
+     |    GET /versions for uuid |                           |
+     o-------------------------->|                           |
+     |<-----/200 json payload/---o                           |
+     |                           |                           |
+     |            200 with computed JSON update manifest     |
+     |            indicating what files should               |
+     |            be downloaded                              |
+     o------------------------------------------------------>|
+     |                           |                           |
+----
+
+[[update-manifest]]
+=== Update Manifest
+
+The responses sent to the client must be well-formed JSON documents, referred
+to as "update manifests" which the client must understand. The Update Manifest
+should have a consistent structure but will be dynamically generated _per
+client_ in order to ensure that the client is only downloading what is
+necessary to update that specific client.
+
+.Example Update Manifest
+[source,json]
+----
+{
+    "core" : {
+        "uri" : "https://updates.jenkins.io/some/path/to/a/jenkins.war",
+        "checksum" : {
+            "type" : "sha256",
+            "signature" : "somechecksumforthefile"
+        }
+    },
+    "plugins" : {
+        "updates" : [
+            {
+                "uri" : "https://updates.jenkins.io/some/path/to/a/plugin.hpi",
```

Comment:
`updates.jenkins.io` as currently implemented cannot serve versioned content, much less commits from JEP-305. Can we serve from Artifactory (perhaps via repo.azure.jenkins.io proxy)?

Reply:
I'll replace with `example.com` domains for clarity. 

updates.jenkins.io is just an Apache server, so yes it can serve up whatever we put there :smiling_imp: